weather/api_reader.py

```diff
@@ -0,0 +1,47 @@
+import requests
+import pandas as pd
+from datetime import datetime, timedelta
+
+def getYearData():
+    
+    """
+    reads in the API request and uses f-strings to adjust the start date and end date for collecting necessary data.
+
+    the output should be a pandas dataframe containing both the date (format YYYY-MM-DD) and the daily avg temp in Fahrenheit
+
+    dateStart and dateEnd are adjusted after each execution of the for loop due and are re-inserted into the api key parameters
+    """
+    
+    API_url = 'http://api.weatherapi.com/v1/history.json'
+    API_token = 'b5d1d68681064ffcb53173100230706'
+    
+    #Populates the current day and sets a datestart 365 days in the past and sets an end date 30 days after due to api 
+    #restrictions set to gathering only ~30 days of data at a time 
+    currDate = datetime.date(datetime.now())
+    dateStart = currDate - timedelta(days=365)
+    dateEnd = dateStart + timedelta(days=30)
+    
+    #init pandas dataframe
+    parameters = ['date', 'avgtemp_f']
+    df = pd.DataFrame(columns=parameters)
+
+    for i in range(0, 13):
+        result = requests.get(f'{API_url}?key={API_token}&q=Houston&dt={dateStart}&end_dt={dateEnd}')
+        result_dict = dict(result.json())
+        if result_dict == None: break
+
+        #For loop checks the dictionary for the necessary values and stores them into a new_row dataframe, it is then
+        #transposed and concat'd with the pre-existing dataframe
+        for i in range(len(result_dict['forecast']['forecastday'])):
+            date = result_dict['forecast']['forecastday'][i]['date']
+            avgTempF = result_dict['forecast']['forecastday'][i]['day']['avgtemp_f']
+            
+            new_row = pd.DataFrame([date, avgTempF], index=parameters).T
+            df = pd.concat([df,new_row], ignore_index=True)
+
+        dateStart = dateEnd + timedelta(days=1)
+        dateEnd = dateStart + timedelta(days=30)
+
+        if dateEnd >= datetime.date(datetime.now()): dateEnd = datetime.date(datetime.now())
+        
+    return df
```

Comment:
Thanks for the feedback. I'll read up on it and share it with the other guys. 

For the pagination, I read through the [documentation of the weatherapi API](https://www.weatherapi.com/docs/), and there no was mention of pages or pagination (only that for pulling historical data, you are limited to pulling 30 days worth of data at a time). As for the for loop, due to the limitations on the amount of days I could query, and the parameters in the api_url needing to be updated, I could not find a more simple solution; however, I'll do some research/tinkering around with my code to see if I can find a better way of doing it.

I'll give them a heads up so we merge our two files.

Reply:
So you don't need to review with Gabe since the class has already been merged with main. You need to pull in the changes if you have not already, and then add your function to the class. For the for loop, let's just leave it as is for now. Since the API does not support pagination, I don't think you will find a much more elegant solution than just running through it with a for loop.